src/main/java/lv/ctco/javaschool/app/boundary/TripApi.java

```diff
@@ -74,13 +56,40 @@ public ListTripDto getActiveTrips() {
         return listTripDto;
     }
 
+    private TripDto convertToTripDto(Trip trip) {
+        User driver = trip.getDriver();
+        TripDto dto = new TripDto();
+        dto.setId(trip.getId());
+        dto.setDriverInfo(driver.getSurname() + " " + driver.getName());
+        dto.setDriverPhone(driver.getPhoneNumber());
+        dto.setEvent(trip.isEvent());
+        dto.setFrom(trip.getDeparture());
+        dto.setTo(trip.getDestination());
+        dto.setPlaces(trip.getPlaces());
+        dto.setTime(trip.getDepartureTime());
+        dto.setTripStatus(trip.getTripStatus());
+        return dto;
+    }
+
     @POST
     @Path("/{id}")
     @RolesAllowed({"ADMIN", "USER"})
-    public void setTripPlaces(JsonObject field, @PathParam("id") String tripId) {
-        for (Map.Entry<String, JsonValue> pair : field.entrySet()) {
-            Integer places = ((JsonNumber) pair.getValue()).intValue();
-            tripStore.setTripPlaces(places, tripId);
+    public Response setTripPlacesAndUser(JoinTripDto joinTripDto, @PathParam("id") Long tripId) {
+        User user = userStore.getCurrentUser();
+        Optional<Trip> tripOptional = tripStore.findTripById(tripId);
+        if (tripOptional.isPresent()) {
+            Trip trip = tripOptional.get();
+            if (trip.getPassengers().contains(user)) {
+                throw new ValidationException("The user have already joined this trip");
+            } else {
+                List<User> passengers = trip.getPassengers();
+                passengers.add(user);
+                trip.setPassengers(passengers);
+                trip.setPlaces(joinTripDto.getPlaces() - 1);
+                return Response.status(Response.Status.OK).build();
```

Comment:
Now you can just skip that line and change return type to `void`.

Reply:
changed return type of the setTripPlacesAndUser() method to void